docs/administrators/operational_procedures.rst

```diff
@@ -0,0 +1,216 @@
+Operational Procedures
+----------------------
+
+This document describes the best practices for various operation procedures.
+
+.. note::
+    issue templates are available for all procedures at the bottom of this page
+
+
+TODO: process flowchart
+
+Project Release for Production
+###############################
+
+This process describes the steps to prepare an inmanta project for production release.
+
+Context
+++++++++
+* The project development and testing is complete
+* All modules have been properly released.
+* The project is in a git repo, with a specific branch dedicated to production releases
+* The project is checked out on disk.
+* All modules are checked out on the correct, tagged commit.
+
+Procedure
+++++++++++
+
+1. Verify in `project.yml` that `install_mode` is set to `release`.
+2. (Optional) set the `version` attribute in `project.yml` to the version of this release.
+3. Freeze all modules with
+
+.. code-block:: bash
+
+    inmanta -vv -X project freeze --recursive --operator "=="
+
+3. Verify that all modules are frozen to the correct version
+4. Commit this change (`git commit -a`)
+5. (Optional) Tag this commit (`git tag -a $version`)
+6. Push to the release branch (`git push`)
+
+Upgrade of service model on the orchestrator
+#############################################
+
+This process describes how to safely take an existing project from one version to the next.
+
+Context
+++++++++
+* The orchestrator has the project already deployed and running
+* The project is released (as described above)
+
+Pre-Upgrade steps
+++++++++++++++++++
+1. Verify that environment safety setting are on (this should always be the case)
+
+    * purge_on_delete = False
+    * protected_environment = True
+
+2. Temporarily disable auto_deploy
+
+   * auto_deploy = False
+
+3. Click ‘recompile’ to verify that no new deploy would start.
+
+    * A new version will appear but it will not start to deploy
+
+4. Inspect the current state of the latest active version, verify no failures are happening and the deploy looks healthy
+5. (Optional) Perform a dryrun. Wait for the dryrun to complete and take note of all changes detected by the dryrun. Ideally there should be none.
+
+Upgrade procedure
+++++++++++++++++++
+1. Click `Update and recompile`
+
+   * A new version will appear but it will not start to deploy
+
+2. Click dryrun on the new version
+
+   * The dryrun report will open
+   * Wait for the dryrun to finish
+   * Inspect any changes found by the dryrun, determine if they are expected. If unexpected things are present, go to the abort procedure.
+3. If all is OK, click deploy to make the changes effective
+
+Post Upgrade procedure
++++++++++++++++++++++++++
+
+1. Re-enable auto_deploy
+
+    * Auto_deploy = True
```

Comment:
click force deploy?

Reply:
that should not be required, but I can add it?